WordPress/gutenberg
# Lodash: Refactor away from `_.first()` and `_.last()`

## What?
This PR removes Lodash's `_.first()` and `_.last()` completely and deprecates them. There are just a few usages and conversion is pretty straightforward. 

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

We're replacing a few usages with their native counterparts. 

## Testing Instructions

* Verify the block appender after the last block still looks and works alright. 
* Verify indenting and outdenting in a list still work the same way. Make sure to test outdent with the enter key.
* Smoke test global styles.
* Verify all checks are green - some of the changes are covered by tests.
